homeassistant/components/transmission/sensor.py

```diff
@@ -56,30 +64,20 @@ class TransmissionSensor(SensorEntity):
 
     _attr_should_poll = False
 
-    def __init__(self, tm_client, client_name, sensor_name, sub_type=None):
+    def __init__(self, tm_client, client_name, sensor_name, key):
         """Initialize the sensor."""
         self._tm_client: TransmissionClient = tm_client
-        self._client_name = client_name
-        self._name = sensor_name
-        self._sub_type = sub_type
+        self._attr_name = f"{client_name} {sensor_name}"
```

Comment:
Please don't add this unwanted composing of name with device name.
Instead, leave it as is and in another PR set `_attr_has_entity_name = True` to have the name composing happen automatically in frontend.

Reply:
I'm not adding name composing, I'm keeping existing functionality just moving it from function to attribute.

I've been explicitly asked to avoid doing too many things in one PR so this one is focused on unique_id. Once it's merged I'll be happy to migrate to new entity naming.

Reply:
You're right, sorry about that!